packages/utils/src/deviceUtils.ts

```diff
@@ -0,0 +1,72 @@
+// define required device attributes here to avoid dependencies
+// can be replaced in the future when we create a device types package
+type PartialDevice = Partial<{
+    features: {
+        bootloader_mode: boolean | null;
+        bootloader_hash: string | null;
+        revision: string | null;
+        major_version: number | null;
+        minor_version: number | null;
+        patch_version: number | null;
+        fw_major: number | null;
+        fw_minor: number | null;
+        fw_patch: number | null;
+        initialized: boolean | null;
+        no_backup: boolean | null;
+    };
+    firmwareType: string | null;
+}>;
+
+export const getDeviceMode = (device?: PartialDevice) => {
+    if (device?.features?.bootloader_mode) return 'bootloader';
+    if (!device?.features?.initialized) return 'initialize';
+    if (device?.features?.no_backup) return 'seedless';
+    return 'normal';
+};
+
+export const isDeviceInBootloader = (device?: PartialDevice) => !!device?.features?.bootloader_mode;
+
+export const getFwRevision = (device?: PartialDevice) => device?.features?.revision || '';
+
+export const getBootloaderHash = (device?: PartialDevice) =>
+    device?.features?.bootloader_hash || '';
+
+export const getBootloaderVersion = (device?: PartialDevice) => {
+    if (!device?.features) {
+        return '';
+    }
+    const { features } = device;
+
+    if (isDeviceInBootloader(device) && features.major_version) {
+        return `${features.major_version}.${features.minor_version}.${features.patch_version}`;
+    }
+
+    return '';
+};
+
+export const getFwVersion = (device?: PartialDevice) => {
```

Comment:
Why not rename it to `getFirmwareVersion` for better readability?

Reply:
I will rename all the methods you mentioned.

2. It is not true anymore. There is  a versionUtils, xssFilters and bufferUtils. It was once discussed on standup or coinjoin meeting with szymon.

Reply:
Yes, it is true that some of the files broke that "rule".
But what do you  think? Do you think using a file per function improves developer experience or it does not?

Reply:
done